### PR TITLE
[ZT] Generic s2s

### PR DESCRIPTION
**Pull request by ranbel**
PCX-11136

### CODE REVIEW

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+### 3. Add a service provider integration
+
+To create a custom service-to-service integration:
+
+<Render file="posture/add-service-provider" params={{ one: "Custom service provider" }} />
+5. In **Access client ID** and **Access client secret**, enter the Access service token used to authenticate to your external API.
+6. In **Rest API URL**, enter the external API endpoint that Cloudflare will query for posture information (for example, `https://api.example.com`). For more information, refer to [External API requirements](#external-api-requirements).
+7. In **Polling frequency**, choose how often Cloudflare Zero Trust should query the external API for information.
+8. Select **Test and save**.
```
Comment (ranbel) [inline src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx]: What are we actually testing?

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx
Comment (kokolocomotion1) [inline src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx:1]: looks good just need a change in the 'device posture attributes' section to show a value between 0-100

Comment (kokolocomotion1) [reply src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx:1]: couple more additional things
1. can you rename this page as 'Custom device posture integration' so it better matches our UI and terraform documentation?
2. also please make sure this page is hidden from the nav for now. after we're through with more beta testing we can unhide it.

Comment (ranbel) [reply src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx:1]: > looks good just need a change in the 'device posture attributes' section to show a value between 0-100

I believe this is already covered in https://github.com/cloudflare/cloudflare-docs/pull/16698/files#diff-8d2277e76b27bc5387bd23dd8a551ea877e3e529115053a6172da7cc6dfdcd81R126. Or are you referring to a different part of the docs?

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+### 4. Configure the posture check
+
+<Render file="posture/configure-posture-check" params={{ one: "Custom service provider" }} />
+
+## Device posture attributes
+
+| Selector      | Description                                                                                       | Value                                                                                         |
+| ------------- | ------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------- |
+| Score            | Posture score returned by external API                                                                                   | `1` to `100`                                                                                  |
```
Comment (kokolocomotion1) [inline src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx]: hey Ranbel this is the section I'm referring to where the value is shown as 1-100 but it should be 0-100

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx
Comment (kokolocomotion1) [reply src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx:1]: I left a comment in the section I was referring to

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx (hunk 2 above)
Comment (ranbel) [reply src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx]: good catch

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
+---
+pcx_content_type: how-to
+title: Custom device posture integration
+sidebar:
+  label: Custom integration
+  order: 1
+  hidden: true
```
Comment (marciocloudflare) [inline src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx:7]: Just to double check you really want to hide this page from navigation, and if so is there a link to it?

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+---
+
+import { Render } from "~/components"
+
+Cloudflare Zero Trust allows you to enforce custom device posture checks on your applications. This involves configuring a WARP service-to-service integration that periodially calls the external API of your choice, whether it is a third-party endpoint provider or a home built solution. When called, the API will receive device identifying information from Cloudflare and be expected to return a value between 0 to 100. You can then set up a device posture check that determines if the returned value counts as a pass or fail; for example, you could allow access to a user only if their device has a posture value greater than 60.
```
Comment (marciocloudflare) [inline src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx]:
```suggestion
Cloudflare Zero Trust allows you to enforce custom device posture checks on your applications. This involves configuring a WARP service-to-service integration that periodially calls the external API of your choice, whether it is a third-party endpoint provider or a home built solution. When called, the API will receive device identifying information from Cloudflare and be expected to return a value between `0` to `100`. You can then set up a device posture check that determines if the returned value counts as a pass or fail; for example, you could allow access to a user only if their device has a posture value greater than `60`.
```

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+The custom service provider integration works with any API service that meets the following specifications.
+
+### Authentication
+
+The WARP client authenticates to the external API through Cloudflare Access. The external API should [validate the application token](/cloudflare-one/identity/authorization-cookie/validating-json/) issued by Cloudflare Access to ensure that any requests which bypass Access (for example, due to a network misconfiguration) are rejected.
+
+### Data passed to external API
+
+Cloudflare will pass the following parameters to the configured API endpoint. You can use this data to identify the device and assign a posture score. For some devices, not all identifying information will apply, in which case the field will be blank. A maximum of 1000 devices will be sent per a request.
```
Comment (marciocloudflare) [inline src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx]:
```suggestion
Cloudflare will pass the following parameters to the configured API endpoint. You can use this data to identify the device and assign a posture score. For some devices, not all identifying information will apply, in which case the field will be blank. A maximum of 1,000 devices will be sent per a request.
```

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+| Field | Description |
+| ----- | ----------- |
+| device_id | Device UUID assigned by the WARP client |
+| email | Email address used to authenticate the WARP client |
+| serial_number | Device serial number |
+| mac_address | Device MAC address |
+| virtual_ipv4 | Device virtual IPv4 address |
+| hostname | Device name |
```
Comment (marciocloudflare) [inline src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx]:
```suggestion
| `device_id` | Device UUID assigned by the WARP client |
| `email` | Email address used to authenticate the WARP client |
| `serial_number` | Device serial number |
| `mac_address` | Device MAC address |
| `virtual_ipv4` | Device virtual IPv4 address |
| `hostname` | Device name |
```

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+```
+
+### Expected response from external API
+
+For each Cloudflare `device_id`, the API service is expected to return a posture score and optionally a third-party device ID.
+
+| Field | Description |
+| ----- | ----------- |
+| s2s_id | Third party device ID (empty string if unavailable) |
+| score | Integer value between 0 - 100 |
```
Comment (marciocloudflare) [inline src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx]:
```suggestion
| `s2s_id` | Third party device ID (empty string if unavailable) |
| `score` | Integer value between `0` - `100` |
```

File: src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx (hunk 3 above)
Comment (ranbel) [reply src/content/docs/cloudflare-one/identity/devices/service-providers/custom.mdx:7]: Yes, we want to hide it from the nav for now but have it display in- dash.

File: src/content/partials/cloudflare-one/posture/add-service-provider.mdx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+inputParameters: param1
```
Comment (marciocloudflare) [inline src/content/partials/cloudflare-one/posture/add-service-provider.mdx]: Consider start using the new way for parameters in variables https://developers.cloudflare.com/style-guide/components/render/#partials

**Review by marciocloudflare — APPROVED** [review]

**Comment by bllchmbrs** [comment]
@ranbel , Hyperlint dev here. The GitHub API had an intermittent API failure. You can push a nominal change to run the checks again or just ignore them.

**Comment by bllchmbrs** [comment]
Well of course it recovers from that failure right when I post the comment.

**Review by kokolocomotion1 — APPROVED** [review]

**Pull request merged**